ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/certificates/CertificatePinner.kt

```diff
@@ -436,3 +430,5 @@ public data class CertificatePinner(
         )
     }
 }
+
+public class CertificatePinnerException(message: String) : IOException(message)
```

Comment:
I have concerns regarding this exception:
- Previously `DarwinHttpRequestException` was thrown for such cases, so it is potentially breaking change
- This exception is used for any error thrown by the pinner. Should we give a possibility to distinct between these errors?
- Should we provide a multiplatform exception to be able to handle SSL pinning problems in a unified manner?

Reply:
@osipxd Thanks for this PR! I have very little context on the Ktor code, so I don't have much feedback to share. I can see that the exception contains the output message which is exactly what I was hoping to be able to access. I also appreciate how you included some tests. Thanks for your help and @e5l thank you for your review as well! 🙏